core/src/HIP/Kokkos_HIP_GraphNodeKernel.hpp

```diff
@@ -130,9 +129,9 @@ class GraphNodeKernelImpl<Kokkos::HIP, PolicyType, Functor, PatternTag, Args...>
   auto get_driver_storage() const { return m_driver_storage; }
 
  private:
-  Kokkos::ObservingRawPtr<const hipGraph_t> m_graph_ptr    = nullptr;
-  Kokkos::ObservingRawPtr<hipGraphNode_t> m_graph_node_ptr = nullptr;
-  mutable std::shared_ptr<base_t> m_driver_storage         = nullptr;
+  hipGraph_t const* m_graph_ptr                    = nullptr;
+  hipGraphNode_t* m_graph_node_ptr                 = nullptr;
```

Comment:
These are opaque pointer types. My understanding of wrapping them in `ObservingRawPtr` is that they wanted to better tell the reader if the class has to manage the resource (node and graph) or not.

If we're getting rid of `ObservingRawPtr`, I'm not sure we need pointers to opaque pointers.

I'm fine doing it in a followup.

```suggestion
  hipGraphNode_t m_graph_node_ptr                 = nullptr;
```

Reply:
Follow up is fine.